services/src/main/java/org/keycloak/services/clientregistration/oidc/DescriptionConverter.java

```diff
@@ -326,7 +323,7 @@ public static OIDCClientRepresentation toExternalResponse(KeycloakSession sessio
             }
         }
 
-        if (client.getClientAuthenticatorType().equals(ClientIdAndSecretAuthenticator.PROVIDER_ID)) {
+        if (!client.getClientAuthenticatorType().equals(JWTClientAuthenticator.PROVIDER_ID)) {
```

Comment:
This solution is not flexible enough as there are other client authenticators (like for example `X509ClientAuthenticator`), which should not include secrets. Also secret should not be included for public clients (case when `none` is used as client authentication method).

I suggest something bit more flexible like:
- Introduce new method on `ClientAuthenticatorFactory` interface like:
``` 
boolean default supportsSecret() {
     return false;
}
```
And override that just for ClientIdAndSecretAuthenticator and for JWTClientSecretAuthenticator.

- Move the check inside this "else" block https://github.com/keycloak/keycloak/blob/main/services/src/main/java/org/keycloak/services/clientregistration/oidc/DescriptionConverter.java#L322-L326 . Note that you already have ClientAuthenticatorFactory here. So perhaps something like this will do the trick:
```
if (clientAuth.supportsSecret() {
            response.setClientSecret(client.getSecret());
            response.setClientSecretExpiresAt(
                    OIDCClientSecretConfigWrapper.fromClientRepresentation(client).getClientSecretExpirationTime());
}
```

Is it fine for you to update like this?

Reply:
Hi, Thanks for your suggestion.
It's conducive to me.
I totally agree to update it.